### PR TITLE
Update Sentiment Analysis tutorial for ML .NET .7

### PR DESCRIPTION
**Pull request by JRAlexander**
Doc revision is at https://github.com/dotnet/docs/pull/8839

### CODE REVIEW

File: machine-learning/tutorials/SentimentAnalysis/Program.cs [hunk 1]
```diff
@@ -3,102 +3,131 @@
...
+        }
+
+        public static ITransformer Train(MLContext mlContext, string dataPath)
+        {
+            //Note that this case, loading your training data from a file, 
+            //is the easiest way to get started, but ML.NET also allows you 
+            //to load data from databases or in-memory collections.
             // <Snippet6>
-            pipeline.Add(new TextLoader(_dataPath).CreateFrom<SentimentData>());
+            IDataView dataView =_textLoader.Read(new MultiFileSource(dataPath));
```
Comment (shmoradims) [inline machine-learning/tutorials/SentimentAnalysis/Program.cs]: >new MultiFileSource(dataPath) [](start = 49, length = 29)

let's use dataPath directly for the simple case of loading from a single path. MultiFileSource is not needed. Applies to most cases.

File: machine-learning/tutorials/SentimentAnalysis/Program.cs [hunk 2]
```diff
@@ -3,102 +3,131 @@
...
 
-            // TextFeaturizer is a transform that is used to featurize an input column. 
-            // This is used to format and clean the data.
+            // Create a flexible pipeline (composed by a chain of estimators) for creating/training the model.
+            // This is used to format and clean the data.  
+            // Convert the text column to numeric vectors (Features column) 
             // <Snippet7>
-            pipeline.Add(new TextFeaturizer("Features", "SentimentText"));
-            //</Snippet7>
+            var pipeline = new TextFeaturizingEstimator(mlContext,"SentimentText", "Features")
```
Comment (shmoradims) [inline machine-learning/tutorials/SentimentAnalysis/Program.cs]: >new TextFeaturizingEstimator(mlContext [](start = 27, length = 38)

replace w/ mlContext.Transforms.Text.FeaturizeText. Please prefer mlContext extensions when available.

File: machine-learning/tutorials/SentimentAnalysis/Program.cs [hunk 3]
```diff
@@ -3,102 +3,131 @@
...
+             //</Snippet7>
 
-            // Adds a FastTreeBinaryClassifier, the decision tree learner for this project, and 
-            // three hyperparameters to be used for tuning decision tree performance.
-            // <Snippet8>
-            pipeline.Add(new FastTreeBinaryClassifier() { NumLeaves = 50, NumTrees = 50, MinDocumentsInLeafs = 20 });
+            // Adds a FastTreeBinaryClassificationTrainer, the decision tree learner for this project  
+            // <Snippet8> 
+                    .Append(mlContext.BinaryClassification.Trainers.FastTree(label: "Label", features: "Features", numLeaves: 50, numTrees: 50, minDatapointsInLeafs: 20));
```
Comment (shmoradims) [inline machine-learning/tutorials/SentimentAnalysis/Program.cs]: >label: "Label", features: "Features" [](start = 77, length = 36)

These are default values. What's the best practice for docs? Leave them out, or spell them out?

File: machine-learning/tutorials/SentimentAnalysis/Program.cs [hunk 4]
```diff
@@ -3,102 +3,131 @@
...
         }
 
-        public static void Evaluate(PredictionModel<SentimentData, SentimentPrediction> model)
+        public static void Evaluate(MLContext mlContext, ITransformer model)
         {
-            // Evaluates.
+            // Evaluate the model and show accuracy stats
+            // Load evaluation/test data
+            // <Snippet12>
+            IDataView dataView = _textLoader.Read(new MultiFileSource(_testDataPath));
```
Comment (shmoradims) [inline machine-learning/tutorials/SentimentAnalysis/Program.cs]: >new MultiFileSource(_testDataPath) [](start = 50, length = 34)

same here.

File: machine-learning/tutorials/SentimentAnalysis/Program.cs [hunk 5]
```diff
@@ -3,102 +3,131 @@
...
+            // </Snippet12>
+
+            //Take the data in, make transformations, output the data. 
             // <Snippet13>
-            var testData = new TextLoader(_testDataPath).CreateFrom<SentimentData>();
+            Console.WriteLine("=============== Evaluating Model accuracy with Test data===============");
+            var predictions = model.Transform(dataView);
```
Comment (shmoradims) [inline machine-learning/tutorials/SentimentAnalysis/Program.cs]: >var [](start = 12, length = 3)

What's the docs best practice for var? Leave it as var, or specify the type explicitly for better readability without IDE?

File: machine-learning/tutorials/SentimentAnalysis/Program.cs [hunk 6]
```diff
@@ -111,57 +140,129 @@ public static void Evaluate(PredictionModel<SentimentData, SentimentPrediction>
...
+            // </Snippet31>          
+        }
+
+        // Saves the model we trained to a zip file.
+
+        private static void SaveModelAsFile(MLContext mlContext, ITransformer model)
+        {
+            // <Snippet24> 
+            using (var fs = new FileStream(_modelPath, FileMode.Create, FileAccess.Write, FileShare.Write))
+                model.SaveTo(mlContext, fs);
```
Comment (shmoradims) [inline machine-learning/tutorials/SentimentAnalysis/Program.cs]: >model.SaveTo [](start = 16, length = 12)

mlContext.Model.Save

File: machine-learning/tutorials/SentimentAnalysis/Program.cs (hunk 3 above)
Comment (JRAlexander) [reply machine-learning/tutorials/SentimentAnalysis/Program.cs]: I've removed them.

**Pull request merged**